HISTORY.md

```diff
@@ -51,9 +54,8 @@
 * Removes non-default parameter in tiledb_config_unset. [#2099](https://github.com/TileDB-Inc/TileDB/pull/2099)
 
 ### C++ API
-* Add new api,  to get a query's config. [#2167](https://github.com/TileDB-Inc/TileDB/pull/2167)
+* Addition of Array::set_config to directly assign a config to an array. Also added an Array constructor that accepts the config and a config getter. [#2142](https://github.com/TileDB-Inc/TileDB/pull/2142)
```

Comment:
Let's just document the new `Array::set_config`. Ideally, we want these history notes to be as direct as possible. I would expect just a list of new APIs here, we don't need to explain what they do.